### ISSUE TITLE
JUnitReportReporter reports skipped tests as failed

### ISSUE THREAD

**Issue by tnaroska** [cbeust/testng#1078]
JUnitReportReporter does not report skipped tests correctly. Instead, skipped tests are reported as failure.
See code here:
https://github.com/cbeust/testng/blob/master/src/main/java/org/testng/reporters/JUnitReportReporter.java#L100-L132

**Comment by juherr** [comment]
:+1: Could you propose a pull-request?

For reference: http://svn.apache.org/repos/asf/ant/core/trunk/src/main/org/apache/tools/ant/taskdefs/optional/junit/XMLJUnitResultFormatter.java

**Comment by tnaroska** [comment]
I can try to submit something. I was also wondering about the relation between
[JUnitReportReporter](https://github.com/cbeust/testng/blob/master/src/main/java/org/testng/reporters/JUnitReportReporter.java) and [JUnitXMLReporter](https://github.com/cbeust/testng/blob/master/src/main/java/org/testng/reporters/JUnitXMLReporter.java).
Both generate junit xml reports although with some slight differences. e.g. JUnitXMLReporter _does_ report skipped tests correctly.

**Comment by juherr** [comment]
I didn't find any usage of `JUnitXMLReporter` except in a test...

@cbeust Do you remember why there are 2 JUnit reporters? What is your recommandation about them?

**Comment by cbeust** [comment]
One of them is supposed to replace the other but I can't remember which
one... The history might give us a clue, but it looks like JUnitXmlReporter
might be the obsolete one, although we should probably understand why it
reports skipped tests correctly and carry this over in JUnitReportReporter.

-- 
Cédric

**Comment by tnaroska** [comment]
I tried to figure out the history of the two classes, but couldn't really make sense of it. The code got moved around in the repo making it hard to follow the full history.
Anyway, I have a fix for JUnitReportReporter in place and opened PR-1080.

**Comment by juherr** [comment]
Closed by #1080

**Comment by martoe** [comment]
@tnaroska @juherr The PR doesn't correctly resolve this issue: now the tests are reported as "skipped" (which is correct), but the test report contains no error at all.
I guess the correct behaviour would be to additionally include the failed configuration method in the report (like the HTML reporter does)

**Comment by juherr** [comment]
@martoe IMO, it is another issue. Could you open a new one?
